### ISSUE TITLE
Show "Appears On" section for artists

### ISSUE THREAD

**Issue by Chaphasilor** [jmshrv/finamp#572]
Jellyfin supports differentiating between albums where the artist is also an album artists and albums/compilations where the artist simply has a song within an album released by another artist.  
This differentiation is very helpful when trying to manage multiple artists per track, as well as dealing with release compilation albums (e.g. Monstercat, or soundtrack albums).

Screenshot:
![image](https://github.com/jmshrv/finamp/assets/18015147/bd8182de-21fc-40cd-8a2e-7d11ec91d140)

Finamp should support and show this section as well, so that one can still link artists with all the albums they appear on, but without requiring them to be an actual album artist for that album.  
This would also be helpful for downloading artist collections, since all direct albums could be fully downloaded, and for albums where they only appear on the metadata and their own tracks would be downloaded, but not the tracks by other artists.

**Comment by bsfreq** [comment]
This would be very cool. 
I also wish you can add the option to easily add all artist songs (including appears on songs/albums) to next up / queue / playlist etc. 
I often search for an artist and I’d like to be able to shuffle all artist songs+appers on songs/albums to queue. Currently I have to tap+hold each search result separately. Perhaps you could add a new search result (Artist + appears on) that we can tap+hold to easily add all those to queue?

**Comment by Chaphasilor** [comment]
Well right now you can add all artist album tracks to the queue, this would probably simply be extended to include the tracks from appears on as well...  
Or do you mean something else?

**Comment by bsfreq** [comment]
> ll artist album tracks to the queue, this would probably simply be extended to include the tracks from appears on as well...
> Or do you mean something else?

This exactly. An easy way to add everything from an artist, including appears on tracks..

**Comment by ravtakhar** [comment]
The image linked above is the artists page. This would be very cool on the Finamp artist page allowing users to quickly navigate between all of their songs.

**Comment by lukaslindnermusic** [comment]
I will include this in PR https://github.com/jmshrv/finamp/pull/1145 😊

**Comment by bsfreq** [comment]
> I will include this in PR [#1145](https://github.com/jmshrv/finamp/pull/1145) 😊

Beautiful, thank you so much!

**Comment by lukaslindnermusic** [comment]
No problem 😊 
Will look like this:
![Image](https://github.com/user-attachments/assets/c1c85381-03cc-43a3-8bb2-4493a4406fc9)

If there are a lot of albums, the Appears On section obviously disappears down the list.
What do you think would be a good solution to reach it quickly? 
Maybe collapsing the lists by pressing on the Titles (so a tap on "Albums" would hide all albums)?

**Comment by bsfreq** [comment]
Yes, perhaps have the lists open by default, but have arrows to show that they are open and collapsible.
Something like this:

![Image](https://github.com/user-attachments/assets/d8debe96-024c-4868-a1e6-f0e585e8e950)

**Comment by Chaphasilor** [comment]
When the artist screen is properly redesigned we will probably have horizontally-scrolling sections with the first 10 albums or so for each section, and then a "see all" button that takes you to a discography page or similar.  
Could also work for tracks, to easily see all tracks (in order of release). That's an issue I had with Spotify, because it only offers browsing artist discography by album, not by track

**Comment by lukaslindnermusic** [comment]
> When the artist screen is properly redesigned we will probably have horizontally-scrolling sections with the first 10 albums or so for each section, and then a "see all" button that takes you to a discography page or similar. Could also work for tracks, to easily see all tracks (in order of release). That's an issue I had with Spotify, because it only offers browsing artist discography by album, not by track

Yeah, that sounds nice! 😊 
But for this PR I think it's best to wrap it up. I think, for now I would add the collapsing titles as suggested by @bsfreq as a temporary solution. Is that ok? 😊

**Comment by Chaphasilor** [comment]
Added in [beta update 0.9.16](https://github.com/jmshrv/finamp/releases/tag/0.9.16-beta)!
